wondrous-app/utils/board/dataFetching.tsx

```diff
@@ -1,43 +1,66 @@
 import apollo from 'services/apollo';
-import { GET_TASK_BY_ID, GET_TASK_PROPOSAL_BY_ID, GET_PREVIEW_FILE } from 'graphql/queries';
+import {
+  GET_TASK_BY_ID,
+  GET_TASK_PROPOSAL_BY_ID,
+  GET_PREVIEW_FILE,
+  GET_GRANT_BY_ID,
+  GET_GRANT_APPLICATION_BY_ID,
+} from 'graphql/queries';
 
 // eslint-disable-next-line import/prefer-default-export
 export const getServerSideProps = async (context) => {
-  if (context.query?.task || context.query?.taskProposal) {
+  if (context.query?.task || context.query?.taskProposal || context.query?.grant || context.query?.grantApplicationId) {
     try {
       const meta = {
         title: '',
         description: '',
         img: '',
       };
 
-      let task;
+      let entity;
 
       if (context.query?.taskProposal) {
         const { data } = await apollo.query({
           query: GET_TASK_PROPOSAL_BY_ID,
           variables: { proposalId: context.query?.taskProposal },
         });
 
-        task = data.getTaskProposalById;
-      } else {
+        entity = data.getTaskProposalById;
+      }
+      if (context.query?.task) {
         const { data } = await apollo.query({
           query: GET_TASK_BY_ID,
           variables: { taskId: context.query.task },
         });
 
-        task = data.getTaskById;
+        entity = data.getTaskById;
       }
 
-      if (!task) {
+      if (context.query?.grant && !context.query?.grantApplicationId) {
+        const { data } = await apollo.query({
+          query: GET_GRANT_BY_ID,
+          variables: { grantId: context.query.grant },
+        });
+
+        entity = data.getGrantById;
+      }
+
+      if (context.query?.grantApplicationId) {
+        const { data } = await apollo.query({
+          query: GET_GRANT_APPLICATION_BY_ID,
+          variables: { grantApplicationId: context.query.grantApplicationId },
+        });
+        entity = data.getGrantApplicationById;
+      }
+      if (!entity) {
         return { props: {} };
       }
 
-      meta.title = task.title;
-      meta.description = JSON.parse(task.description)[0]?.children[0]?.text;
```

Comment:
@Lamperoyge does this work for null description?

Reply:
true, I'll put fallback value